Code/src/main/java/nl/utwente/group10/haskell/type/ListTypeCon.java

```diff
@@ -0,0 +1,20 @@
+package nl.utwente.group10.haskell.type;
+
+import java.util.ArrayList;
+
+public class ListTypeCon extends TypeCon {
+
+    ListTypeCon() {
+        super("[]");
+    }
+
+    @Override
+    public String asTypeAppChain(final int fixity, final ArrayList<Type> args)
+    {
+        if (args.size() != 1) {
```

Comment:
The typeparser doesn't support it yet, but   ([a])  ([] a)  ([])  are all valid Haskell types.